diff --git a/advanced_forum.module b/advanced_forum.module
index 1891545..2450d08 100644
--- a/advanced_forum.module
+++ b/advanced_forum.module
@@ -857,7 +857,7 @@ function advanced_forum_statistics_online_users() {
 function advanced_forum_get_reply_link($node) {
   $reply_link = array();
 
-  $fragment = $node->content['comments']['comment_form']['#id'];
+  $fragment = 'comment-form';
 
   if ($comment_setting == COMMENT_NODE_OPEN) {
 
