diff --git a/ubc_healthcheck.module b/ubc_healthcheck.module
index 71ab392..178a300 100644
--- a/ubc_healthcheck.module
+++ b/ubc_healthcheck.module
@@ -88,6 +88,16 @@ function ubc_healthcheck_scom() {
 
 }
 
+/*
+ * Disable password reset links 
+ */
+function ubc_healthcheck_mail_alter(&$message) {
+
+        if (false != stripos($message['subject'], "Replacement login information for")) {
+          $message['body'] = str_replace("http", "hxxp", $message['body']);
+        }
+}
+
 /*
  * Test the DB connection by querying the node table
  */
@@ -240,4 +250,4 @@ function _ubc_healthcheck_insertStatusString($status_array) {
 		if(!is_numeric($status)) return STATUS_ERR;
 	}
 	return STATUS_OK;
-}
\ No newline at end of file
+}
