diff --git a/redirect_after_login.module b/redirect_after_login.module
index 5949b65..bf9ee28 100644
--- a/redirect_after_login.module
+++ b/redirect_after_login.module
@@ -93,6 +93,20 @@ function redirect_after_login_user_login($account) {
     // We want to redirect user on login.
     $current_user = Drupal::currentUser();
     $username = $current_user->getRoles();
+
+    $roles = $username;
+    // if there are only two roles 'CWL' (and 'authenticated') roles then exclude from redirection
+    // as we want user to be returned to requested URL as per ReturnTo param
+    \Drupal::logger('redirect_on_login')->notice("roles: " . print_r($roles, TRUE));
+    if(2 == count($roles)
+    && in_array('cwl', $roles)
+    && in_array('authenticated', $roles)) {
+      \Drupal::logger('redirect_on_login')->notice("Don't redirect - CWL and authenticated role only");
+      return;
+    } else {
+      \Drupal::logger('redirect_on_login')->notice("Redirect - not CWL user and/or has multiple roles");
+    }
+
     $login_redirection = Drupal::config('redirect_after_login.settings')
       ->get('login_redirection');
     if (isset($login_redirection[array_reverse($username)[0]])) {
