diff --git a/conditional_fields.module b/conditional_fields.module
index 58a8374..3bfc524
--- a/conditional_fields.module
+++ b/conditional_fields.module
@@ -245,6 +245,10 @@ function conditional_fields_element_after_build($element, &$form_state) {
     $entity_type = $field['#entity_type'];
     $bundle = $field['#bundle'];
   }
+  elseif (isset($field[0]['#entity_type'], $field[0]['#bundle'])) {
+    $entity_type = $field[0]['#entity_type'];
+    $bundle = $field[0]['#bundle'];
+  }
   elseif (isset($form['#entity_type'], $form['#bundle'])) {
     $entity_type = $form['#entity_type'];
     $bundle = $form['#bundle'];
