diff --git a/simplesamlphp_auth.module b/simplesamlphp_auth.module
index b888814..abcba12 100644
--- a/simplesamlphp_auth.module
+++ b/simplesamlphp_auth.module
@@ -295,8 +295,12 @@ function simplesamlphp_auth_form_alter(&$form, $form_state, $form_id) {
 
   $login_path = variable_get('simplesamlphp_auth_login_path', 'saml_login');
   $login_name = variable_get('simplesamlphp_auth_login_link_display_name', t('Federated Log In'));
+  $login_image = path_to_theme() . '/img/cwl-login-btn.gif';
+
   $destination = drupal_get_destination();
-  $link = l($login_name, $login_path, array('query' => $destination));
+  $link = l($login_image, $login_path, array(
+    'query' => $destination,
+    'html' => TRUE));
 
   // Add SAML login link to user login form.
   if ($form_id == 'user_login_block' || $form_id == 'user_account_form') {
@@ -306,7 +310,9 @@ function simplesamlphp_auth_form_alter(&$form, $form_state, $form_id) {
   }
 
   if ($form_id == 'user_login') {
+    $form['links']['#prefix'] = '<div class="cwl-login">';
     $form['links']['#markup'] = $link;
+    $form['links']['#suffix'] = '</div>';
   }
 
   if (($form_id == 'user_register_form' || $is_user_profile_account_form) && user_access('change saml authentication setting')) {
