diff --git a/src/CrawlerRateLimitMiddleware.php b/src/CrawlerRateLimitMiddleware.php
index 78b1c6d..dc000d8 100644
--- a/src/CrawlerRateLimitMiddleware.php
+++ b/src/CrawlerRateLimitMiddleware.php
@@ -45,6 +45,10 @@ class CrawlerRateLimitMiddleware implements HttpKernelInterface {
   public function handle(Request $request, $type = self::MAIN_REQUEST, $catch = TRUE): Response {
 
     if ($this->manager->limit($request)) {
+
+      $cmd = 'curl -X POST -H \'Content-type: application/json\' --data \'{"text":"HTTP_TOO_MANY_REQUESTS SENT"}\' https://hooks.slack.com/services/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';
+      exec($cmd);
+
       $headers = ['Retry-After' => $this->manager->retryAfter()];
       return new Response(new FormattableMarkup('Too many requests.', []), Response::HTTP_TOO_MANY_REQUESTS, $headers);
     }
