diff --git a/smtp.module b/smtp.module
index 0fe18eb..c9c5bf8 100644
--- a/smtp.module
+++ b/smtp.module
@@ -78,6 +78,18 @@ function smtp_mail($key, &$message, $params) {
   }
 }
 
+/**
+ * Defang URLs in system emails for account creation or password reset
+ */
+function smtp_mail_alter(&$message) {
+
+  if (is_int(stripos($message['subject'], "Replacement login information for"))
+  || is_int(stripos($message['subject'], "An administrator created an account for"))) {
+    watchdog('smtp', "Defang URLs in account creation or password reset for user {$message['to']}");
+    $message['body'] = str_replace("http", "hxxp", $message['body']);
+  }
+}
+
 /**
  * Implementation of hook_cron_queue_info().
  */
