diff --git a/smtp.phpmailer.inc b/smtp.phpmailer.inc
index 95420d2..525e5e4 100644
--- a/smtp.phpmailer.inc
+++ b/smtp.phpmailer.inc
@@ -187,26 +187,26 @@ class PHPMailer {
    * Sets the default SMTP server port.
    * @var int
    */
-  public $Port          = 25;
+  public $Port          = 587;
 
   /**
    * Sets the SMTP HELO of the message (Default is $Hostname).
    * @var string
    */
-  public $Helo          = '';
+  public $Helo          = 'mail.ubc.ca';
 
   /**
    * Sets connection prefix.
    * Options are "", "ssl" or "tls"
    * @var string
    */
-  public $SMTPSecure    = '';
+  public $SMTPSecure    = 'tls';
 
   /**
    * Sets SMTP authentication. Utilizes the Username and Password variables.
    * @var bool
    */
-  public $SMTPAuth      = FALSE;
+  public $SMTPAuth      = TRUE;
 
   /**
    * Sets SMTP username.
@@ -225,7 +225,7 @@ class PHPMailer {
    * This function will not work with the win32 version.
    * @var int
    */
-  public $Timeout       = 10;
+  public $Timeout       = 20;
 
   /**
    * Sets SMTP class debugging on or off.
