diff --git a/ubc_saml_auth.module b/ubc_saml_auth.module
index 992da0a..15240b0 100644
--- a/ubc_saml_auth.module
+++ b/ubc_saml_auth.module
@@ -133,7 +133,11 @@ function ubc_saml_auth_node_access($node, $op, $account) {
           watchdog('UBC_SAML_AUTH', 'drupal_goto: '.$_SERVER['SCRIPT_NAME']);
           
           if (!path_is_admin(current_path())) {
-            drupal_goto('saml_login', array('query' => array('ReturnTo' => $rtn_url)));
+
+            // don't redirect for drush
+            if (! (function_exists('drush_main') && drupal_is_cli())) {
+              drupal_goto('saml_login', array('query' => array('ReturnTo' => $rtn_url)));
+            }
           }
           else {
             watchdog('UBC_SAML_AUTH', 'Is Admin Path');
